bin/get_target_taxa_from_taxid.py

```diff
@@ -0,0 +1,180 @@
+
+Usage:
+    get_target_taxa_from_taxid.py --taxid <taxid> --rankedlineage <path> --level <level> --output <output_file>
+
+Author: GitHub Copilot
+Version: 1.0.0
+"""
+
+import argparse
+import sys
+import os
+
+
+VERSION = "1.0.0"
+
+# Mapping of taxonomy levels to their column indices in rankedlineage.dmp
+# Format: taxid | name | species | genus | family | order | class | phylum | kingdom | domain
+TAXONOMY_LEVELS = {
+    'species': 2,
+    'genus': 3,
+    'family': 4,
+    'order': 5,
+    'class': 6,
+    'phylum': 7,
+    'kingdom': 8,
+    'domain': 9
+}
+
+
+def parse_rankedlineage(rankedlineage_path, target_taxid):
+    """
+    Parse the rankedlineage.dmp file and find the lineage for the given taxid.
+
+    Args:
+        rankedlineage_path: Path to the rankedlineage.dmp file
+        target_taxid: Target taxid to search for
+
+    Returns:
+        Dictionary with taxonomy information or None if not found
+    """
+    if not os.path.isfile(rankedlineage_path):
+        sys.stderr.write(
+            f"ERROR: The NCBI rankedlineage.dmp file was not found at: {rankedlineage_path}\n"
+        )
+        sys.exit(1)
+
+    with open(rankedlineage_path, 'r') as f:
+        for line in f:
+            split_line = line.strip().split('|')
+            split_line = [n.strip() for n in split_line]
+
+            # Validate line format
+            if len(split_line) < 10:
+                continue
+
+            taxid = split_line[0]
+            if taxid == target_taxid:
+                return {
+                    'taxid': split_line[0],
+                    'name': split_line[1],
+                    'species': split_line[2],
+                    'genus': split_line[3],
+                    'family': split_line[4],
+                    'order': split_line[5],
+                    'class': split_line[6],
+                    'phylum': split_line[7],
+                    'kingdom': split_line[8],
+                    'domain': split_line[9]
+                }
+
+    return None
+
+
+def get_target_taxa(lineage_dict, taxonomy_level):
+    """
+    Extract target taxa from lineage dictionary at specified taxonomy level.
+
+    Args:
+        lineage_dict: Dictionary with taxonomy information
+        taxonomy_level: Taxonomy level to extract (e.g., 'order', 'class')
+
+    Returns:
+        String in format 'level:taxa_name' or None if level is empty
+    """
+    if taxonomy_level not in TAXONOMY_LEVELS:
+        sys.stderr.write(
+            f"ERROR: Invalid taxonomy level '{taxonomy_level}'. "
+            f"Valid levels are: {', '.join(TAXONOMY_LEVELS.keys())}\n"
+        )
+        sys.exit(1)
+
+    taxa_name = lineage_dict.get(taxonomy_level, '')
+
+    # Check if the taxa name is empty or just whitespace
+    if not taxa_name or taxa_name.strip() == '':
+        return None
+
+    return f"{taxonomy_level}:{taxa_name}"
+
+
+def main():
+    parser = argparse.ArgumentParser(
```

Comment:
Please pull parse args into it's own function, there should be examples in the bin folder where I try to make sure everything has it's own parse_args function.